### ISSUE TITLE
Cancelling a stalled retrieval-transfer makes the command hang indefinately

### ISSUE THREAD

**Issue by rjan90** [filecoin-project/lotus#7540]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [lotus forum](https://github.com/filecoin-project/lotus/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to lotus.

### Lotus component

- [ ] lotus daemon - chain sync
- [ ] lotus miner - mining and block production
- [ ] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [ ] lotus miner/market - storage deal
- [X] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### Lotus Version

```text
Daemon:  1.13.0-rc3+mainnet+git.c23cff45f+api1.2.0
Local: lotus-miner version 1.13.0-rc3+mainnet+git.c23cff45f
```

### Describe the Bug

When trying to cancel a retrieval transfer that has stalled, the `lotus-miner data-transfers cancel ....xxx` command hangs indefinitely.

It seems to happen most frequently on transfers that has started, and then stalled:
```
Sending Channels

ID                   Status           Sending To   Root Cid     Initiated?  Transferred  Voucher                                   
1634579593870108898  Ongoing          ...m5rYc8VD  ...3zpbzada  N           31.94MiB     ...lIncrease":1048576,"UnsealPrice":"0"}  
  Message: stream reset
1634580045873407428  Ongoing          ...m5rYc8VD  ...3zpbzada  N           362.5MiB     ...lIncrease":1048576,"UnsealPrice":"0"}  
  Message: stream reset
1634597923690491082  Ongoing          ...m5rYc8VD  ...5ozef35m  N           21.99GiB     ...lIncrease":1048576,"UnsealPrice":"0"}  
  Message: stream reset
```
Or transfers that has a `responder paused` status:
```
1634601155040510814  ResponderPaused  ...m5rYc8VD  ...lxnyjiqu  N           0B           ...lIncrease":1048576,"UnsealPrice":"0"}  
  Message: stream reset
```
On transfers that had 0B transferred and `Message: stream reset` I was able to cancel 30+, except 1 that was hanging indefinitaly.

It also causes the markets process to hang if you try to shut it down, and you have to kill the hanging processes:
```
2021-10-19T13:17:19.999+0200  WARN    builder node/shutdown.go:31     received shutdown       {"signal": "interrupt"}
2021-10-19T13:17:20.000+0200    WARN    builder node/shutdown.go:36     Shutting down...
2021-10-19T13:17:20.001+0200    INFO    builder node/shutdown.go:44     rpc server shut down successfully 
2021-10-19T13:17:20.020+0200    INFO    dagstore        dagstore/wrapper.go:406 will close the dagstore
2021-10-19T13:17:20.020+0200    INFO    dagstore        dagstore@v0.4.3/handlers.go:59  failure handler: stopping
2021-10-19T13:17:20.020+0200    INFO    dagstore        dagstore@v0.4.3/dagstore_control.go:48  dagstore closed
2021-10-19T13:17:20.020+0200    INFO    dagstore        dagstore/wrapper.go:410 dagstore closed
2021-10-19T13:17:20.020+0200    INFO    dagstore        dagstore/wrapper.go:413 waiting for dagstore background wrapper goroutines to exit
2021-10-19T13:17:20.020+0200    INFO    dagstore        dagstore/wrapper.go:415 exited dagstore background wrapper goroutines
2021-10-19T13:17:20.020+0200    INFO    dt-impl impl/impl.go:170        stop data-transfer module
2021-10-19T13:17:38.602+0200    INFO    net/identify    identify/id.go:372      failed negotiate identify protocol with peer    {"peer": "12D3KooWCGwpwo5xdcja8jDiHQZArqbe2JBhUfJ4CEdNVAupqx1Q", "error": "stream reset"}
2021-10-19T13:17:58.928+0200    INFO    net/identify    identify/id.go:372      failed negotiate identify protocol with peer    {"peer": "12D3KooWGthHBQfPUuBZzxRoaCwipooGiEo81GCCeqVqHyuk3NZW", "error": "stream reset"}
^C^Z[1]   Killed                  lotus-miner run

[2]+  Stopped                 lotus-miner run
```
If you try to start the markets-process again, you get an error that it already is running:
```
2021-10-19T13:19:24.996+0200    INFO    main    lotus-miner/run.go:100  Checking full node sync status
Worker: 206967; Base: 0; Target: 0 (diff: 0)
State: idle; Current Epoch: 0; Todo: 0

Done!
ERROR: repo is already locked (lotus daemon already running)
```
Killing the hanging transfers helps shut the process down:
```
[3]   Killed                  lotus-miner data-transfers cancel 1634600071306806820
[4]   Killed                  lotus-miner data-transfers cancel 1634600071306805711
[5]   Killed                  lotus-miner data-transfers cancel 1634597923690491082
[6]   Killed                  lotus-miner data-transfers cancel 1634580045873407428
[7]   Killed                  lotus-miner data-transfers cancel 1634579593870108898
[8]-  Killed                  lotus-miner data-transfers cancel 1634600071306805711
[9]+  Killed                  lotus-miner data-transfers cancel 1634597923690491082
```

### Logging Information

```text
lotus-miner log list:
data_transfer_network
peermgr
messagepool
market_adapter
quic-transport
bs:sprmgr
gs-unverifiedbs
panic-reporter
miner
retrievalmarket_impl
markets-rtvl-reval
fullnode
retrievaladapter
partialfile
chainxchg
ffiwrapper
statetree
routing/record
main
repo
dht
wallet-ledger
storageminer
swarm2
tarutil
cliutil
preseal
filestore
routedhost
auth
sectors
dt-impl
dt_graphsync
storagemarket_impl
storageadapter
hello
tracing
modules
blockservice
cmds
fil-consensus
paych
bs:sess
storagemarket_network
gs-asyncloader
pubsub
ffi-wrapper
rand
messagesigner
wallet
mocknet
importmgr
actors
ipns
peerstore
nat
storedask
markets
dt-chanmon
dagstore/upgrader
pathresolv
amt
beacon
splitstore
bs:peermgr
retrieval
p2pnode
blockstore
types
advmgr
net/identify
alerting
table
fsjournal
markets-rtvl
tcp-tpt
storagemrkt
cborrrpc
gen
retrieval-fsm
system
peerstore/ds
badger
statemgr
retrieval_network
events
graphsync_allocator
autorelay
client
net/conngater
evtsm
lock
paramfetch
engine
eventlog
sectoraccessor
diversityFilter
chain
gs_request_executor
vm
ulimit
stream-upgrader
blankhost
unixfs
builder
discovery
disputer
dagstore
drand
build
backupds
chainstore
sub
piecestore
watchdog
cmds/http
dht.pb
bitswap
fsutil
fsm
mplex
boguskey
genesis
providerstates
graphsync
reuseport-transport
basichost
connmgr
metrics
rpcenc
providers
dht/RtRefreshManager
autonat
p2p-config
node
addrutil
payment-channel-settler
rpc
badgerbs
incrt
data-transfer
graphsync_network
gs-traversal
test-logger
stores
ping
retrieval-discovery
cli
bitswap_network
relay
chunk
conngater
```

### Repo Steps

1. Get retrieval deals that stalls
2. When trying to cancel them, the `lotus-miner data-transfers cancel` command hangs indefinitely 
...

**Comment by rjan90** [comment]
Also seeing this issue on all transfers that fails with a `message queue shutdown`:
```
1635531773340554828  Ongoing          ...TEufRsC9  ...N4E9Jx9E  N           556.1MiB     ...lIncrease":1048576,"UnsealPrice":"0"}  
  Message: message queue shutdown
1635531996920972641  Ongoing          ...TEufRsC9  ...N4E9Jx9E  N           31.77MiB     ...lIncrease":1048576,"UnsealPrice":"0"}  
  Message: message queue shutdown
```

**Comment by s0nik42** [comment]
Yes same for me 

lotus-miner  data-transfers  list -v
```
Sending Channels
ID                   Status           Sending To                                            Root Cid                                                     Initiated?  Transferred  Voucher
1635544969849181861  Ongoing          12D3KooWGy8jFJobk3yVH3S63heuijt89Fzq1e75DYtcm5rYc8VD  bafybeihrnnkbsrilzax5r7tkcybpxf7wm62r54xu5d7gasxmy3f5l7masi  N           8.495MiB     {"PayloadCID":{"/":"bafybeihrnnkbsrilzax5r7tkcybpxf7wm62r54xu5d7gasxmy3f5l7masi"},"ID":508199,"Selector":{"Raw":"oWFSomFsoWRub25loGI6PqFhYaFhPqFhQKA="},"PieceCID":null,"PricePerByte":"0","PaymentInterval":1048576,"PaymentIntervalIncrease":1048576,"UnsealPrice":"0"}
  Message: stream reset
1635544969849181866  Ongoing          12D3KooWGy8jFJobk3yVH3S63heuijt89Fzq1e75DYtcm5rYc8VD  bafybeiebvkybbsthf25jn37sdninsvu4p4lkeavd7kuab57o2h7xkyvzle  N           0B           {"PayloadCID":{"/":"bafybeiebvkybbsthf25jn37sdninsvu4p4lkeavd7kuab57o2h7xkyvzle"},"ID":348057,"Selector":{"Raw":"oWFSomFsoWRub25loGI6PqFhYaFhPqFhQKA="},"PieceCID":null,"PricePerByte":"0","PaymentInterval":1048576,"PaymentIntervalIncrease":1048576,"UnsealPrice":"0"}
  Message: stream reset
1635544969849181871  Ongoing          12D3KooWGy8jFJobk3yVH3S63heuijt89Fzq1e75DYtcm5rYc8VD  bafybeihn7kke5a2ukuvuf6hbmxizno3c7vmsq6y7kgzxftbrwiti5aeg4e  N           0B           {"PayloadCID":{"/":"bafybeihn7kke5a2ukuvuf6hbmxizno3c7vmsq6y7kgzxftbrwiti5aeg4e"},"ID":198814,"Selector":{"Raw":"oWFSomFsoWRub25loGI6PqFhYaFhPqFhQKA="},"PieceCID":null,"PricePerByte":"0","PaymentInterval":1048576,"PaymentIntervalIncrease":1048576,"UnsealPrice":"0"}
  Message: stream reset
1635544969849181873  Ongoing          12D3KooWGy8jFJobk3yVH3S63heuijt89Fzq1e75DYtcm5rYc8VD  bafybeihjl2i7hv7p5cjtdixebtgz46yp6dux3luptkdn7fj5hwn6bg7omi  N           0B           {"PayloadCID":{"/":"bafybeihjl2i7hv7p5cjtdixebtgz46yp6dux3luptkdn7fj5hwn6bg7omi"},"ID":195012,"Selector":{"Raw":"oWFSomFsoWRub25loGI6PqFhYaFhPqFhQKA="},"PieceCID":null,"PricePerByte":"0","PaymentInterval":1048576,"PaymentIntervalIncrease":1048576,"UnsealPrice":"0"}
  Message: stream reset
1635544969849181888  Ongoing          12D3KooWGy8jFJobk3yVH3S63heuijt89Fzq1e75DYtcm5rYc8VD  bafybeieuj73azneo6pzds3sgpic3w5h2cex2t35yveum56c3kr5xv6v4cu  N           0B           {"PayloadCID":{"/":"bafybeieuj73azneo6pzds3sgpic3w5h2cex2t35yveum56c3kr5xv6v4cu"},"ID":396666,"Selector":{"Raw":"oWFSomFsoWRub25loGI6PqFhYaFhPqFhQKA="},"PieceCID":null,"PricePerByte":"0","PaymentInterval":1048576,"PaymentIntervalIncrease":1048576,"UnsealPrice":"0"}
  Message: stream reset
1635544969849181974  Ongoing          12D3KooWGy8jFJobk3yVH3S63heuijt89Fzq1e75DYtcm5rYc8VD  bafybeigwdidn3efcnljil23lxjtciobkffjove56xyfmiolgxcstxq6apa  N           0B           {"PayloadCID":{"/":"bafybeigwdidn3efcnljil23lxjtciobkffjove56xyfmiolgxcstxq6apa"},"ID":257846,"Selector":{"Raw":"oWFSomFsoWRub25loGI6PqFhYaFhPqFhQKA="},"PieceCID":null,"PricePerByte":"0","PaymentInterval":1048576,"PaymentIntervalIncrease":1048576,"UnsealPrice":"0"}
  Message: stream reset
1635544969849181989  Ongoing          12D3KooWGy8jFJobk3yVH3S63heuijt89Fzq1e75DYtcm5rYc8VD  bafybeidxi42rl22xcpzmeu7j2krdzdvg6keoqkobnmhaldosipugwgt3ba  N           0B           {"PayloadCID":{"/":"bafybeidxi42rl22xcpzmeu7j2krdzdvg6keoqkobnmhaldosipugwgt3ba"},"ID":886762,"Selector":{"Raw":"oWFSomFsoWRub25loGI6PqFhYaFhPqFhQKA="},"PieceCID":null,"PricePerByte":"0","PaymentInterval":1048576,"PaymentIntervalIncrease":1048576,"UnsealPrice":"0"}
  Message: stream reset
1635544969849181995  Ongoing          12D3KooWGy8jFJobk3yVH3S63heuijt89Fzq1e75DYtcm5rYc8VD  bafybeih2xqem46saysmhwxqrhbzxkm6ud7bqrnjej2zkf42b2pbtwgo2yy  N           0B           {"PayloadCID":{"/":"bafybeih2xqem46saysmhwxqrhbzxkm6ud7bqrnjej2zkf42b2pbtwgo2yy"},"ID":320565,"Selector":{"Raw":"oWFSomFsoWRub25loGI6PqFhYaFhPqFhQKA="},"PieceCID":null,"PricePerByte":"0","PaymentInterval":1048576,"PaymentIntervalIncrease":1048576,"UnsealPrice":"0"}
  Message: stream reset
1635544969849182116  ResponderPaused  12D3KooWGy8jFJobk3yVH3S63heuijt89Fzq1e75DYtcm5rYc8VD  bafybeifurexb5bik6qfjw7oid6jqqeiynfvut74jz7hweff7ph26e3qqxa  N           0B           {"PayloadCID":{"/":"bafybeifurexb5bik6qfjw7oid6jqqeiynfvut74jz7hweff7ph26e3qqxa"},"ID":1096446,"Selector":{"Raw":"oWFSomFsoWRub25loGI6PqFhYaFhPqFhQKA="},"PieceCID":null,"PricePerByte":"0","PaymentInterval":1048576,"PaymentIntervalIncrease":1048576,"UnsealPrice":"0"}
  Message: stream reset
1635544969849182147  Ongoing          12D3KooWGy8jFJobk3yVH3S63heuijt89Fzq1e75DYtcm5rYc8VD  bafybeiebvkybbsthf25jn37sdninsvu4p4lkeavd7kuab57o2h7xkyvzle  N           7.182MiB     {"PayloadCID":{"/":"bafybeiebvkybbsthf25jn37sdninsvu4p4lkeavd7kuab57o2h7xkyvzle"},"ID":672873,"Selector":{"Raw":"oWFSomFsoWRub25loGI6PqFhYaFhPqFhQKA="},"PieceCID":null,"PricePerByte":"0","PaymentInterval":1048576,"PaymentIntervalIncrease":1048576,"UnsealPrice":"0"}
  Message: stream reset
1635544969849182175  ResponderPaused  12D3KooWGy8jFJobk3yVH3S63heuijt89Fzq1e75DYtcm5rYc8VD  bafybeid3s5ahl42iu3rhyugme4a43flq4uyderap4w44dtn2hgydbcva4u  N           0B           {"PayloadCID":{"/":"bafybeid3s5ahl42iu3rhyugme4a43flq4uyderap4w44dtn2hgydbcva4u"},"ID":925021,"Selector":{"Raw":"oWFSomFsoWRub25loGI6PqFhYaFhPqFhQKA="},"PieceCID":null,"PricePerByte":"0","PaymentInterval":1048576,"PaymentIntervalIncrease":1048576,"UnsealPrice":"0"}
  Message: stream reset
```
Command :
`lotus-miner  data-transfers cancel 1635544969849182147`

**Comment by j1v** [comment]
I can confirm the same bug on my side. 

I haven't restarted lotus-miner yet. 

Also, all cancel processes need to be killed

**Comment by rjan90** [comment]
Closing this issue as this seems to be fixed on the newer release candidates for 1.13.2. The cancel command does not hang on release-candidate 2 and up 😄
